kedro/framework/cli/cli.py

```diff
@@ -210,6 +210,8 @@ def main(
                 sys.exit(exc.code)
         except Exception as error:
             logger.error(f"An error has occurred: {error}")
```

Comment:
This message was introduced in 0.19.7; however, if we want to spit out the full trace, why is this necessary? Why can't you just doing something like

```python
            self._cli_hook_manager.hook.after_command_run(
                project_metadata=self._metadata, command_args=args, exit_code=1
            )
            hook_called = True
            raise
```

?

This would be closer to the previous behavior.

In case you're unsure, your `finally` block will still be called:

```pycon
>>> try:
...    print("do or do not")
...    raise Exception("oh no")
... except Exception as exc:
...    print("there is no try")
...    raise
... finally:
...    print("rip")
... 
do or do not
there is no try
rip
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
Exception: oh no
>>> 
```

Reply:
That does output the stack trace but IMO is not as aesthetically nice. This is what it looks like using `raise`:

![image](https://github.com/user-attachments/assets/29ad0a6b-53dc-4cc8-bc7a-4ab1c1240a45)

This is what it looks like using `traceback.format_exc`

![image](https://github.com/user-attachments/assets/c05c0f7a-2e7a-48aa-83ce-018eb34bc2f9)

The content of the trace is the same, but I think the second is easier to read.

Reply:
Another option would be using the rich console traceback, but that would require importing rich.

![image](https://github.com/user-attachments/assets/871ca40c-fa3f-4e62-bf42-1b2760afadd0)

Reply:
> Another option would be using the rich console traceback, but that would require importing rich.

I'm not sure if this is what you mean, but you get the Rich traceback as long as Rich is installed—you don't need to change the code.

In fact, one downside of catching and printing the formatted exception yourself is that you don't get the Rich traceback.

Reply:
Without manually logging formatted exception and just using `raise`, it looks like the first image.

Reply:
I prefer the solution @deepyaman proposes. I've compared that with how it looks in `0.19.6` and it's exactly the same. I wouldn't worry too much about the formatting. 

Side note: I don't know what's going on with the rich formatting. I used to have logs formatted like in the ticket description with the rectangle around them (https://github.com/kedro-org/kedro/issues/4073), but now I'm not seeing that.. If it's an issue in general I'd suggest tackling that separately.

Reply:
I wonder if https://github.com/kedro-org/kedro/pull/3682/ has changed stuff.. I noticed that even if I use the default rich handler it doesn't pick it up in the catalog messages.

Reply:
I've also noticed that our little hack of uninstalling rich and downgrading cookiecutter does not work anymore. I can take a look at that.

Reply:
@merelcht #3682 shouldn't change this as it targets at two very specific log message with rich markup.